packages/math-input/src/components/keypad/mobile-keypad.tsx

```diff
@@ -0,0 +1,158 @@
+import {StyleType} from "@khanacademy/wonder-blocks-core";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+import ReactDOM from "react-dom";
+
+import Key from "../../data/keys";
+import {View} from "../../fake-react-native-web/index";
+import {Cursor, KeypadConfiguration, KeyHandler} from "../../types";
+
+import Keypad from "./index";
+
+/**
+ * This is the v2 equivalent of v1's ProvidedKeypad. It follows the same
+ * external API so that it can be hot-swapped with the v1 keypad and
+ * is responsible for connecting the keypad with MathInput and the Renderer.
+ *
+ * Ideally this strategy of attaching methods on the class component for
+ * other components to call will be replaced props/callbacks since React
+ * doesn't support this type of code anymore (functional components
+ * can't have methods attached to them).
+ */
+
+type Props = {
+    onElementMounted?: (arg1: any) => void;
+    onDismiss?: () => void;
+    style?: StyleType;
+};
+
+type State = {
+    active: boolean;
+    keypadConfig?: KeypadConfiguration;
+    keyHandler?: KeyHandler;
+    cursor?: Cursor;
+};
+
+class MobileKeypad extends React.Component<Props, State> {
+    hasMounted = false;
+
+    state: State = {
+        active: false,
+    };
+
+    activate: () => void = () => {
+        this.setState({active: true});
+    };
+
+    dismiss: () => void = () => {
+        this.setState({active: false});
+    };
+
+    configure: (configuration: KeypadConfiguration, cb: () => void) => void = (
+        configuration,
+        cb,
+    ) => {
+        this.setState({keypadConfig: configuration});
+
+        // HACK(charlie): In Perseus, triggering a focus causes the keypad to
+        // animate into view and re-configure. We'd like to provide the option
+        // to re-render the re-configured keypad before animating it into view,
+        // to avoid jank in the animation. As such, we support passing a
+        // callback into `configureKeypad`. However, implementing this properly
+        // would require middleware, etc., so we just hack it on with
+        // `setTimeout` for now.
```

Comment:
I think this reference to "middleware" is now outdated... as I think that refers to Redux.

Reply:
I commented this...comment... and added a ticket to circle back and investigate if we need this hack at all post-Redux: https://khanacademy.atlassian.net/browse/LC-1080